### PR TITLE
Nav Redesign i3 - Update top bar

### PR DESCRIPTION
**Pull request by lsl**
Fixes https://github.com/Automattic/dotcom-forge/issues/8143

## Proposed Changes

- [x] The WP logo will link to the global all sites page
- [x] The site dropdown will link to the dashboard and front-end of the site
- [x] The "+ New" dropdown will behave like core
- [x] The reader, help center, and notification links will behave as expected
- [x] The "Howdy {name} [AVATAR]" link will take you to contain links to both the site profile and the user profile pages (like it does today)
- [x] On hover menu items for visit site / dashboard
- [x] On hover menu items for "+ New" menu
- [x] Hover color / css
- [x] Something with the sizing on "+" icon
- [x] Mobile sizing for icons
- [x] Menu dropdown styling
- [x] Menu dropdown links for "+ New"
- [x] Hover color needs to match core
- [x] Dropdown links should be clickable cursors anywhere on the row, not just the link
- [x] Breakpoints need checking, @ 674 width text shows up where it shouldn't
- [x] Decide and fix which items to hide at smaller breakpoints, core hides the W logo, we can't do that. Probably: Cart and Home icons. Maybe also move help and notifications into the profile icon dropdown?
- [x] Wpcom has "is active" setting to wpcom logo under my home, core doesn't do this
- [x] Profile icon padding/margins
- [x] Mobile: Reader, Cart, help, notification and profile icons need a size increase and alignment check to match the rest of the icons (36-38px). Wordpress logo, home, and + icons are currently matching core. (Note: did something different for each icon but not sure they're pixel perfect matches)
- [x] Desktop: Cart, help, notifications icons need shrinking to match the left
- [x] Submenu items need to support onClick
- [x] Menu links need polish / confirmation / coordination with wp-admin / classic views
- [x] Breakpoints clean up, specifically hamburger menu item not showing below 780 when it collapses the sidebar
- [x] Dropdown menu item content left alignment is off, should align with the menu bar icons above
- [x] Various todo's
- [x] Post editor menu hamburger icon on mobile doesn't load side bar, possibly just needs hiding.
- [x] Help icon should not shrink when below 480px
- [x] Post editor menu on smaller mobile screens replaces "W" logo with left chevron, we should rework this logic to be closer to desktop behavior.
- [x] Wpcom vs wporg logo - ignoring for now.
- [x] Gravatar in profile dropdown to match wp-admin
- [x] On touch should open dropdowns on mobile. Currently acts like on lick instead of on hover. (looking at this here https://github.com/Automattic/wp-calypso/pull/92609 ) 
- [x] We had to re-add the onClick event in `renderMySites` ("W" logo), on mobile, this isn't working quite right, clicking on it expands the sidebar and then redirects you to /sites. We should only expand on the hamburger menu item click. - Fixing later
- [ ] Color scheme issues: Fix first elsewhere? e.g. https://github.com/Automattic/wp-calypso/pull/92621
- [x] e2e Tests are now failing

## Testing Instructions

* [x] All site level calypso screens (/home)
* [x] All global level calypso screens (plugins, reader, /me)
* [x] Calypso green?
* [x] A4A?
* [x] Post editor screens (/post)
* [x] Site editor screens
* [x] Localization (we have min-width on dropdowns)
* [x] Domain only domain sites
* [x] Ecommerce plans
* [x] VIP sites
* [x] Dev docs
* [x] Anywhere else this might load

### CODE REVIEW

File: client/layout/masterbar/item.tsx [hunk 1]
```diff
@@ -21,6 +25,7 @@ interface MasterbarItemProps {
 	children?: ReactNode;
 	alwaysShowContent?: boolean;
 	disabled?: boolean;
+	subItems?: Array< MasterbarSubItemProps >;
```
Comment (lsl) [inline client/layout/masterbar/item.tsx:31]: Nice approach!

**Comment by lsl** [comment]
Current progress
Screenshot 2024-07-09 at 16-48-30 My Home ‹ Site Title — WordPress com
Screenshot 2024-07-09 at 16-50-50 My Home ‹ Site Title — WordPress com

File: client/layout/masterbar/style.scss [hunk 2]
```diff
@@ -252,6 +260,62 @@ body.is-mobile-app-view {
...
+	&.has-subitems {
+		display: block;
+	}
+
+	> a {
+		display: flex;
+		align-items: center;
+		justify-content: center;
+		gap: 5px;
+	}
```
Comment (lsl) [inline client/layout/masterbar/style.scss:275]: May need scoping to .mb

File: client/layout/masterbar/style.scss [hunk 3]
```diff
@@ -302,12 +366,18 @@ body.is-mobile-app-view {
...
+		}
+
+
 		@include breakpoint-deprecated( ">480px" ) {
 			background: var(--color-masterbar-item-hover-background);
 			color: var(--color-masterbar-highlight);
 			.masterbar--is-checkout & {
 				background: var(--color-checkout-masterbar-item-hover-background);
-				color: var(--color-checkout-masterbar-text);
+				color: var(--color-checkout-masterbar-highlight);
```
Comment (lsl) [inline client/layout/masterbar/style.scss]: Scope to .mb

File: client/layout/masterbar/style.scss [hunk 4]
```diff
@@ -689,10 +891,6 @@ body.is-mobile-app-view {
 }
 
 .masterbar__reader {
-	@include breakpoint-deprecated( ">480px" ) {
-		margin-right: auto;
-	}
```
Comment (lsl) [inline client/layout/masterbar/style.scss:694]: Check this didn't break anything in prod / scope to .mb

File: client/layout/masterbar/style.scss (hunk 2 above)
Comment (lsl) [reply client/layout/masterbar/style.scss:275]: Part of subitems changes, menuitems are either links, buttons, or in the case of sub menu menus, divs containing links.

File: client/layout/masterbar/logged-in.jsx [hunk 5]
```diff
@@ -670,6 +797,71 @@ class MasterbarLoggedIn extends Component {
...
+							{ this.renderSearch() }
+						</div>
+						<div className="masterbar__section masterbar__section--right">
+							{ this.renderCart() }
+							{ this.renderCommandPaletteSearch() }
+							{ this.renderLaunchpadNavigator() }
+							{ loadHelpCenterIcon && this.renderHelpCenter() }
+							{ this.renderNotifications() }
+							{ this.renderProfileMenu() }
+						</div>
```
Comment (lsl) [inline client/layout/masterbar/logged-in.jsx]: These two menus match now aside from the render reader false param. todo: consolidate

Comment (Addison-Stavlo) [reply client/layout/masterbar/logged-in.jsx]: should be resolved now with [61f0bee](https://github.com/Automattic/wp-calypso/pull/92472/commits/61f0bee3b814dfd235e0cab84865046d7ef3c384)

**Comment by eoigal** [comment]
Fixed `Dropdown menu item content left alignment is off, should align with the menu bar icons above` (I think!)

## Before

Screenshot 2024-07-10 at 16 45 26

## After

Screenshot 2024-07-10 at 16 46 40

**Comment by Addison-Stavlo** [comment]
Updated the URLs for the +new items [92676b4](https://github.com/Automattic/wp-calypso/pull/92472/commits/92676b42ebc45572f864c3b56680fe2d58e3cd9e)

File: client/layout/masterbar/style.scss (hunk 4 above)
Comment (lsl) [reply client/layout/masterbar/style.scss:694]: It didn't look like it in my testing.

File: client/layout/masterbar/logged-in.jsx [hunk 6]
```diff
@@ -660,6 +805,45 @@ class MasterbarLoggedIn extends Component {
...
+				return (
+					<Masterbar>
+						<div className="masterbar__section masterbar__section--left">
+							{ this.renderMySites() }
+						</div>
+						<div className="masterbar__section masterbar__section--right">
+							{ this.renderCart() }
+							{ this.renderNotifications() }
+						</div>
+					</Masterbar>
```
Comment (lsl) [inline client/layout/masterbar/logged-in.jsx]: Seems buggy at two different mobile breakpoints. Tempted to replace this with the desktop version and just hide certain menu items when in the editor.

**Review by fushar — COMMENTED** [review]
Edit: FIXED ✅ 

On small viewports, when we're in a site-specific hosting dashboard, the W logo should point to `/sites`. Currently it somehow does nothing when clicked:

image

**Comment by eoigal** [comment]
Updated gravatar image to look the same as wp-admin (on WPCOM simple sites) - also added hover style to copy what's done on other topbar menu items

## Before

Screenshot 2024-07-11 at 12 59 18

## After

Screenshot 2024-07-11 at 12 58 44

**Comment by fushar** [comment]
> Updated gravatar image to look the same as wp-admin (on WPCOM simple sites) 

@eoigal Wait, are we sure? In Core, the gravatar image should be square, following Core's. See p1720546443113669-slack-C06DN6QQVAQ

**Comment by fushar** [comment]
> Gravatar in profile dropdown to match wp-admin

@eoigal I think what Louis meant here, we need to add a gravatar image on the left of the submenu items, matching wp-admin:

image

**Comment by eoigal** [comment]
Thanks @fushar - I'll revert that commit. I was looking at simple wp-admin and assumed that was what we were trying to do - thought it was too easy!

**Comment by Addison-Stavlo** [comment]
> On touch should open dropdowns on mobile. Currently acts like on lick instead of on hover.

Ive been looking at this. I opened [a PR](https://github.com/Automattic/wp-calypso/pull/92609) to merge into this one, trying to mimic how core handles this but from within our react components.

Initially I made the new branch b/c I didn't want to flood this PR with stashed WIP commits while figuring things out. At this point I think its wiser to allow folks to review/test and merge that into here, as if I did it myself there is a higher likelyhood of us overlooking something we may be missing in that implementation.

**Comment by eoigal** [comment]
Updated the sizing of the help center and notifications icons.

Both icons look too small to me here;
Screenshot 2024-07-12 at 11 41 33

Now they will look like this;
Screenshot 2024-07-12 at 11 42 06

The help center icon rendered small on mobile resolutions too;
Screenshot 2024-07-12 at 11 40 01

Now the icon should be the correct size;
Screenshot 2024-07-12 at 11 43 02

**Comment by eoigal** [comment]
I've updated the howdy dropdown to look more like core. It still has some style issue that I'm working on but committed what I have in case any else can figure it out (I'm afk next week).

Screenshot 2024-07-12 at 17 23 37
Screenshot 2024-07-12 at 16 57 27

File: client/layout/masterbar/logged-in.jsx [hunk 7]
```diff
@@ -378,6 +344,151 @@ class MasterbarLoggedIn extends Component {
...
+		}
+
+		return (
+			<>
+				<Item
+					className="masterbar__item-my-site"
+					url={ siteUrl }
+					icon={ <span className="dashicons-before dashicons-admin-home" /> }
+					tipTarget="Menu"
+					subItems={ [ { label: translate( 'Visit your site' ), url: siteUrl } ] }
```
Comment (fushar) [inline client/layout/masterbar/logged-in.jsx]: I believe this should be `Visit Site` as this is what Core uses.

```suggestion
					subItems={ [ { label: translate( 'Visit Site' ), url: siteUrl } ] }
```

File: client/layout/masterbar/logged-in.jsx [hunk 8]
```diff
@@ -653,112 +764,64 @@ class MasterbarLoggedIn extends Component {
...
-			);
-		}
-
-		if ( this.props.isUnifiedSiteView ) {
+		// Editor specific masterbar, only shows back to home button and help center as these are hidden on mobile views
+		// from the desktop version of the editor.
+		// The desktop version of the editor has no masterbar at all so we only do this for mobile.
+		if ( isInEditor && ( isMobile || isResponsiveMenu ) ) {
 			return (
-				<Masterbar className="masterbar__unified">
```
Comment (fushar) [inline client/layout/masterbar/logged-in.jsx:688]: We can now remove the entire `.masterbar__unified` selectors from the CSS file.

File: client/layout/masterbar/logged-in.jsx [hunk 9]
```diff
@@ -653,112 +764,64 @@ class MasterbarLoggedIn extends Component {
...
+				<Masterbar>
 					<div className="masterbar__section masterbar__section--left">
-						{ this.state.isResponsiveMenu
-							? this.renderSidebarMobileMenu()
-							: this.renderWordPressIcon() }
```
Comment (fushar) [inline client/layout/masterbar/logged-in.jsx:692]: Can we now remove this function -- `renderWordPressIcon()` ?

File: client/layout/masterbar/logged-in.jsx [hunk 10]
```diff
@@ -150,11 +158,6 @@ class MasterbarLoggedIn extends Component {
 	clickMySites = () => {
 		this.props.recordTracksEvent( 'calypso_masterbar_my_sites_clicked' );
 
-		if ( this.props.isMobileGlobalNavVisible ) {
```
Comment (fushar) [inline client/layout/masterbar/logged-in.jsx:153]: Can we now remove `isMobileGlobalNavVisible` props?

Comment (lsl) [reply client/layout/masterbar/logged-in.jsx:153]: Along with a few others https://github.com/Automattic/wp-calypso/pull/92472/commits/c9df61847bb4e1408051aab562afb124b605b6ad

File: client/layout/masterbar/logged-in.jsx (hunk 9 above)
Comment (lsl) [reply client/layout/masterbar/logged-in.jsx:692]: Removed https://github.com/Automattic/wp-calypso/pull/92472/commits/cf7925e94b023e33e027c84b17139faf12bb22c8

File: client/layout/masterbar/logged-in.jsx (hunk 8 above)
Comment (lsl) [reply client/layout/masterbar/logged-in.jsx:688]: Global nav too I guess: https://github.com/Automattic/wp-calypso/pull/92472/commits/7c98adc355d1c303bfdcf5e41fd59912c1b8cf47

**Review by fushar — APPROVED** [review]
Tested on various flows on mobile and desktop and they look good. Nice!

**Comment by davemart-in** [comment]
### Non-blocker feedback discovered while testing

Each of these are minor IMO and can be addressed in follow up PRs:

- [ ] On a simple/default site, when I visit the WooCommerce promo page, I don't see the site (house) icon for some reason.

CleanShot 2024-07-16 at 03 16 29

- [ ] We need to add a "Dashboard link to the site (house) icon dropdown. This way, when you are on a global page, you have one-click access back to My Home or wp-admin for the selected site. 

CleanShot 2024-07-16 at 03 21 36

CleanShot 2024-07-16 at 03 29 27

- [ ] Global screen masterbar should still show the shopping cart icon since they are Calypso pages:

CleanShot 2024-07-16 at 03 23 48

**Review by davemart-in — APPROVED** [review]
This is looking solid. I tested it across dozens of different scenarios. There are still minor things to clean up, but we can do those in follow up PRs. Let's get this shipped.

**Comment by fushar** [comment]
>  We need to add a "Dashboard link to the site (house) icon dropdown. This way, when you are on a global page, you have one-click access back to My Home or wp-admin for the selected site.

It should be part of the existing issue:

- https://github.com/Automattic/dotcom-forge/issues/8234

**Comment by tyxla** [comment]
In #97891, I've done a big cleanup that removes much of the code that ended up being unused after this PR. 

I could use your review, folks - thanks!

**Pull request merged**